nat-lab/tests/test_batching.py

```diff
@@ -0,0 +1,281 @@
+                    ConnectionTag.DOCKER_CONE_CLIENT_2,
+                    AdapterType.LinuxNativeWg,
+                    ["stun", "upnp", "local"],
+                    direct_enabled=True,
+                    batching_enabled=True,
+                ),
+                _generate_setup_parameters(
+                    ConnectionTag.DOCKER_UPNP_CLIENT_1,
+                    AdapterType.LinuxNativeWg,
+                    ["stun", "upnp", "local"],
+                    direct_enabled=True,
+                    batching_enabled=True,
+                ),
+                _generate_setup_parameters(
+                    ConnectionTag.DOCKER_UPNP_CLIENT_2,
+                    AdapterType.LinuxNativeWg,
+                    ["stun", "upnp", "local"],
+                    direct_enabled=True,
+                    batching_enabled=True,
+                ),
+            ],
+            id="with_batching",
+        ),
+    ],
+)
+async def test_batching_star(setup_params: List[SetupParameters], request) -> None:
+    async with AsyncExitStack() as exit_stack:
+        capture_duration = 200
+
+        gateways = [DOCKER_GW_MAP[param.connection_tag] for param in setup_params]
+        env = await setup_mesh_nodes(exit_stack, setup_params)
+
+        alpha_client, beta_client, gamma_client, delta_client = env.clients
+        alpha, beta, gamma, delta = env.nodes
+
+        await asyncio.gather(
+            alpha_client.wait_for_state_peer(
+                beta.public_key,
+                [State.Connected],
+                [PathType.Direct],
+            ),
+            alpha_client.wait_for_state_peer(
+                gamma.public_key,
+                [State.Connected],
+                [PathType.Direct],
+            ),
+            beta_client.wait_for_state_peer(
+                alpha.public_key,
+                [State.Connected],
+                [PathType.Direct],
+            ),
+            beta_client.wait_for_state_peer(
+                gamma.public_key,
+                [State.Connected],
+                [PathType.Direct],
+            ),
+            gamma_client.wait_for_state_peer(
+                alpha.public_key,
+                [State.Connected],
+                [PathType.Direct],
+            ),
+            gamma_client.wait_for_state_peer(
+                beta.public_key,
+                [State.Connected],
+                [PathType.Direct],
+            ),
+        )
+
+        gateway_container_names = []
+        for conn_tag in gateways:
+            gateway_container_names.append(container_id(conn_tag))
+
+        conns = [client._connection for client in env.clients]
+        node_container_names = [
+            conn.container_name()
+            for conn in conns
+            if isinstance(conn, DockerConnection)
+        ]
+
+        container_names = gateway_container_names + node_container_names
+        # TODO: so far assuming these are docker containers
+        print("container names: ", container_names)
```

Comment:
Planned diagnostic or leftover?

Reply:
Nice to see which containers are gonna receive `tcpdump` treatment :)